azure-iot-device/setup.py

```diff
@@ -27,13 +27,18 @@
 with open("README.md", "r") as fh:
     _long_description = fh.read()
 
-constant = {}
-with open("azure/iot/device/constant.py") as fh:
-    exec(fh.read(), constant)
+
+version = None
+with open("azure/iot/device/constant.py", "r") as fh:
+    for line in fh:
+        if "VERSION" in line:
+            constant, value = line.strip().split("=")
+            version = value.strip()
+            break
 
```

Comment:
I like it, now what happens if for some reason we remove version from the Constant file.  Will it break or will it fail silently and we not know it?

Reply:
it will fail silently in this case.....for the user agent case it may fail...i have to check that part though

---
In reply to: [389161964](https://github.com/Azure/azure-iot-sdk-python/pull/482#discussion_r389161964) [](ancestors = 389161964)

Reply:
Could we add a check to make sure that the version is not none.

Reply:
+1

---
In reply to: [389205255](https://github.com/Azure/azure-iot-sdk-python/pull/482#discussion_r389205255) [](ancestors = 389205255)

Reply:
I have changed the code a little bit as i realized it would not have covered all bases. Tested with constant file content like :-

```
SOME_VERSION = "322"
IOTHUB_IDENTIFIER = "azure-iot-device-iothub-py"
PROVISIONING_IDENTIFIER = "azure-iot-device-provisioning-py"
VERSION = "2.1.0"
IOTHUB_API_VERSION = "2018-06-30"
PROVISIONING_API_VERSION = "2019-03-31"
SECURITY_MESSAGE_INTERFACE_ID = "urn:azureiot:Security:SecurityAgent:1"
TELEMETRY_MESSAGE_SIZE_LIMIT = 262144
```

---
In reply to: [389206423](https://github.com/Azure/azure-iot-sdk-python/pull/482#discussion_r389206423) [](ancestors = 389206423,389205255)

Reply:
I have changed the code a little bit as i realized it would not have covered all bases. Tested with constant file content like :-

```
SOME_VERSION = "322"
IOTHUB_IDENTIFIER = "azure-iot-device-iothub-py"
PROVISIONING_IDENTIFIER = "azure-iot-device-provisioning-py"
VERSION = "2.1.0"
IOTHUB_API_VERSION = "2018-06-30"
PROVISIONING_API_VERSION = "2019-03-31"
SECURITY_MESSAGE_INTERFACE_ID = "urn:azureiot:Security:SecurityAgent:1"
TELEMETRY_MESSAGE_SIZE_LIMIT = 262144
```

---
In reply to: [389197616](https://github.com/Azure/azure-iot-sdk-python/pull/482#discussion_r389197616) [](ancestors = 389197616,389161964)